kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/RecordEncryptionST.java

```diff
@@ -111,6 +97,12 @@ void ensureClusterHasEncryptedMessage(String namespace) {
                 Constants.KAFKA_DEFAULT_NAMESPACE, numberOfMessages, Duration.ofMinutes(2));
         LOGGER.atInfo().setMessage("Received: {}").addArgument(resultEncrypted).log();
 
+        String messageToCheck = topicName + "vault";
+        if (testKmsFacade.getKmsServiceClass().getSimpleName().toLowerCase().contains("aws")) {
+            messageToCheck = ((AbstractKubeAwsKmsTestKmsFacade) testKmsFacade).getKekKeyId();
+        }
```

Comment:
The need to do this feels wrong. It feels a bit like the contract of `testKekManager.generateKek(topicName);` is wrong and it should be returning the ID of the KEK it generated (though that is potentially a much wider scoped refactor). If it were to do that I think we lose a level of `Abstract*` from the Aws hierarchy. It also feels reasonable for the *test only* API to expose the KEKid as otherwise we have no clue what it did. 

The other alternative I see is we find a different way of confirming the record is encrypted (though the current method also confirms we used the appropriate master key...)

Reply:
Yeah, for me was difficult to fin a proper way to check this. I conclude the same to you regarding the contract of the `generateKek` method, but currently it requires to refactor a lot.

We can create an issue to think about this more carefully.

Reply:
How the filter has encrypted the message is really implementation detail.   The fact that Vault uses a human readable keyname and the AWS currently uses a stringified key id is not this test's concern.  Nor is the fact we write that to the beginning of the record value.

How about just asserting the message the cluster has does not match the plain text?

Reply:
I agree with the fact that checking the content of an encrypted message should not be the test's concern, but by chance we have some readable content that can be checked.

If we don't check that, how can we assure that the content has been encrypted? If we just assert that the plain text is not included in the message, what about if the encryption is broken and just remove the latest character `Hello-worl`? It will be treated as a `Pass`, when it shouldn't.

Reply:
I think it is okay to trust the unit tests are actually ensuring that the KMS is actually doing the encryption right.  We don't need to retest that here.